lib/info.dart

```diff
@@ -46,8 +50,26 @@ abstract class Info {
 // TODO(sigmund): add more:
 //  - inputSize: bytes used in the Dart source program
 abstract class BasicInfo implements Info {
+  static Set<int> _ids = new Set<int>();
   final InfoKind kind;
-  final int id;
+
+  int _id;
+  int get id {
+    if (_id == null) {
+      // we hope we have all of the parents set up by now...
+
+      var path = _getParentPath(this);
```

Comment:
let's reuse some of the utilities we have for this, so instead of `_getParentPath`, use `longName` from `lib/src/utils.dart`?

The one caveat is that longName uses the library.name instead of library.uri, but maybe we can add a similar helper function `uniqueName`?